### DISCUSSION TITLE
Types missing in Theme UI 0.3?

### DISCUSSION THREAD

**Discussion by johnnieskywalker** [system-ui/theme-ui#2205] [Q&A] [answered]
Hi I'm running theme-ui and have following dependencies in package.json
```
    "theme-ui": "^0.3.1",
    "@types/theme-ui": "^0.3.1",
    "@theme-ui/style-guide": "^0.3.5",
    "@types/theme-ui": "^0.3.1",
```
and as suggested in https://theme-ui.com/migrating/ I have no @mdx-js/react.
Everything was working fine, I was not updating theme-ui itself just couple other packages, that seemed unrelated to theme-ui.
And now I'm missing theme-ui types for ex. `TS2305: Module '"theme-ui"' has no exported member 'Heading'.`
I don't know what I'm doing wrong do I miss some theme-ui dependency?

**Comment by hasparus** [comment, marked as answer]
Do you still encounter this error after updating to a newer Theme UI version?

v0.3 was not written in TypeScript, so you might have some problems with it. If this seems to be an issue just on type-level, in `@types/theme-ui`, you could try to publish a fix to DefinitelyTyped.

**Reply by johnnieskywalker** [reply]
Well I tried just by updating like `yarn add @types/theme-ui` and also other theme-ui dependencies. Our project is all in ts and here's my [package.json](https://github.com/OasisDEX/oasis-borrow/blob/561feec63d73ce33c6adb40c82ab1a495b81b304/package.json). May this have something to do with version of typescript or other dependency?

**Reply by hasparus** [reply]
Take a look at `@types/theme-ui` NPM page. There’s a deprecation notice at the top.

New versions of Theme UI are written in TypeScript, and contain their own types.
`@types/theme-ui` never lived in this repo, it was published from DefinitelyTyped.

If you care about TypeScript experience, I’d advise updating to Theme UI 0.14, which is actively maintained and described in the latest docs.

If you can’t update to a version newer than 0.3, and you encounter any mismatch between the types and the runtime, you can copy the types from `@types/theme-ui` to your project and update them there.

Please understand that Theme UI 0.3 was published a few years ago, and we can’t provide non-critical support for it without hurting further development of Theme UI.

**Reply by johnnieskywalker** [reply]
ah ok I thought 0.3 was newer than o.14, my bad

**Reply by johnnieskywalker** [reply]
I can't find version like 0.14 looks like newest is 0.6. I'm confused.
![image](https://user-images.githubusercontent.com/6871923/164729063-06ace560-3190-4a9a-9a32-ade37223be46.png)

**Reply by hasparus** [reply]
No worries! I distinctly remember thinking "Why isn't v1.1.0 the same version v1.10.0?" some time in the past. It can definitely be a bit confusing.

**Reply by johnnieskywalker** [reply]
yeah but whan I add `0.14` in `package.json` then I see this selection during installation, can't pick 0.14

**Reply by hasparus** [reply]
You can't find a newer version of `@types/theme-ui` than 0.6.0, because all Theme UI versions starting from 0.6.0 contain their own types, so the typings library `@types/theme-ui` is redundant and thus deprecated.

If you install `@types/theme-ui` 0.6.0 you should see a message in your terminal, saying this is a stub library, and you can safely remove it.

TLDR Use only `theme-ui`.
